src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts

```diff
@@ -0,0 +1,145 @@
+            delete: "=?",
+            print: "=?",
+            report: "=?",
+            cancel: "=?",
+            edit: "=?"
+        },
+        templateUrl: adhConfig.pkg_path + pkgLocation + "/ResourceActions.html",
+        link: (scope) => {
+            adhPermissions.bindScope(scope, () => scope.resourcePath && AdhUtil.parentPath(scope.resourcePath), "proposalItemOptions");
+        }
+    };
+};
+
+export var reportActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"report();\">{{ \"TR__REPORT\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.report = () => {
+                column.toggleOverlay("abuse");
+            };
+        }
+    };
+};
+
+export var shareActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"report();\">{{ \"TR__SHARE\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.report = () => {
+                column.toggleOverlay("share");
+            };
+        }
+    };
+};
+
+export var deleteActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"delete();\">{{ \"TR__DELETE\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.delete = () => {
+                column.$broadcast("triggerDelete", scope.proposalUrl);
+            };
+        }
+    };
+};
+
+export var printActionDirective = (
+    adhTopLevelState : AdhTopLevelState.Service,
+    $window : Window
+) => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"print();\">{{ \"TR__PRINT\" | translate }}</a>",
+        scope: {
+            class: "@"
+        },
+        link: () => {
+            scope.print = () => {
+                // only the focused column is printed
+                adhTopLevelState.set("focus", 1);
+                $window.print();
+            };
+        }
+    };
+};
+
+export var editActionDirective = (
+    adhTopLevelState : AdhTopLevelState.Service,
+    adhResourceUrl,
+    $location : angular.ILocationService
+) => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"edit();\">{{ \"TR__EDIT\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
```

Comment:
Yes I was just doing these as per needed, was going to add this in future revisions.